packages/extension/src/lib/extensionScripts.ts

```diff
@@ -77,21 +60,39 @@ export const useExtensionPermission = ({
       });
       client.setQueryData(contentScriptKey, true);
       await registerBrowserContentScripts();
-      window.open(companionPermissionGrantedLink, '_blank');
+
+      if (onPermission) {
+        await onPermission(true);
+      } else {
+        window.open(companionPermissionGrantedLink, '_blank');
+      }
+    } else {
+      trackEvent({
+        event_name: 'decline content scripts',
+        extra: JSON.stringify({ origin }),
+      });
```

Comment:
Where was this before?

Reply:
this did not exist in the past, it is a new event being tracked as per spec: https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/578617368/WT-1528+Engagement+loops#Tracking-changes

Updated the analytics taxonomy as well: https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0&range=169:169